packages/app/src/components/time-series-chart/logic/legend.tsx

```diff
@@ -17,16 +19,30 @@ export function useLegendItems<T extends TimestampedValue>(
       shapeComponent: <SeriesIcon config={x} />,
     }));
 
+    /**
+     * Maximum number of legend items
+     *
+     * Used to determine if there are enough items to render a legend
+     */
+    let maxNumItems: number = items.length;
+
     /**
      * Add annotations to the legend
      */
     if (dataOptions?.timespanAnnotations) {
+      maxNumItems += dataOptions?.timespanAnnotations.length;
       for (const annotation of dataOptions.timespanAnnotations) {
-        items.push({
-          label: annotation.label,
-          shape: 'custom',
-          shapeComponent: <TimespanAnnotationIcon />,
-        } as LegendItem);
+        const annotationVisible =
```

Comment:
We definitely intent to prefix all booleans in normal variables.

```suggestion
        const isAnnotationVisible =
```